packages/fields/src/actions/set-as-homepage.tsx

```diff
@@ -0,0 +1,247 @@
+					page_for_posts: 0,
+					page_on_front: 0,
+					show_on_front: 'posts',
+				} );
+
+				createSuccessNotice(
+					__( 'Homepage set to display latest posts' ),
+					{
+						type: 'snackbar',
+					}
+				);
+				onActionPerformed?.( items );
+			} catch ( error ) {
+				const typedError = error as CoreDataError;
+				const errorMessage =
+					typedError.message && typedError.code !== 'unknown_error'
+						? typedError.message
+						: __(
+								'An error occurred while setting the homepage to display latest posts'
+						  );
+				createErrorNotice( errorMessage, { type: 'snackbar' } );
+			}
+		}
+
+		const renderModalBody = () => {
+			if ( 'posts' === showOnFront ) {
+				return (
+					<>
+						<Text>
+							{ sprintf(
+								// translators: %s: title of the page to be set as the homepage.
+								__(
+									'Set "%s" as the site homepage? This will replace the current homepage which is set to display latest posts.'
+								),
+								pageTitle
+							) }
+						</Text>
+					</>
+				);
+			}
+
+			const modalTranslatedString =
+				// translators: %1$s: title of page to be set as the home page. %2$s: title of the current home page.
+				__(
+					'Set "%1$s" as the site homepage? This will replace the current homepage: "%2$s"'
+				);
+
+			return (
+				<Text>
+					{ sprintf(
+						modalTranslatedString,
+						pageTitle,
+						currentHomePageTitle
+					) }{ ' ' }
+				</Text>
+			);
+		};
+
+		const submitAction =
+			showOnFront === 'posts' || item.id !== pageForPosts
+				? onSetPageAsHomepage
+				: onSetLatestPostsHomepage;
+
+		return (
+			<form onSubmit={ submitAction }>
+				<VStack spacing="5">
+					{ renderModalBody() }
+					<HStack justify="right">
+						<Button
+							__next40pxDefaultSize
+							variant="tertiary"
+							onClick={ () => {
+								closeModal?.();
+							} }
+						>
+							{ __( 'Cancel' ) }
+						</Button>
+						<Button
+							__next40pxDefaultSize
+							variant="primary"
+							type="submit"
+						>
+							{
+								// translators: Button to confirm setting the specified page as the homepage.
+								__( 'Set homepage' )
+							}
+						</Button>
+					</HStack>
+				</VStack>
+			</form>
+		);
+	};
+
+const setAsHomepage: Action< PostWithPermissions > = {
+	id: 'set-as-homepage',
+	label: __( 'Set as homepage' ),
+	isEligible( post ) {
+		if ( post.status === 'trash' ) {
+			return false;
+		}
+
+		if ( post.type !== PAGE_POST_TYPE ) {
+			return false;
+		}
+
+		// A front-page template overrides homepage settings, so don't show the action if it's present.
+		const templates = select( coreStore ).getEntityRecords(
+			'postType',
+			'wp_template',
+			{
+				per_page: -1,
+			}
+		);
+		if (
+			templates?.some(
+				( template ) =>
+					'slug' in template && template.slug === 'front-page'
+			)
+		) {
+			return false;
+		}
+
+		const siteSettings = select( coreStore ).getEntityRecord< Settings >(
```

Comment:
I'm not sure we should have subscriptions inside the `isEligible` function.. --cc @youknowriad @oandregal

Reply:
yes, isEligible should use "select" at all.

Reply:
Thanks both. This didn't feel right but I wasn't sure how else to do it. I've tried refactoring this so that both of the `select` calls are outside of the `isEligible` function. I've moved them to the `PostList` and `PostAction` components, so the settings can be passed into the action from higher up.